Add install method from cli in doc (#1538)
PrestaShop/docs: basics/installation/install-from-cli.md

| `email`         | Admin user email                           | pub@prestashop.com           | _string_                                                                                              |
| `license`       | Show PrestaShop license after installation | 0                            | 0, 1                                                                                                  |
| `theme`         | Theme name to install                      | "" (classic)                 | Theme name (located in `/themes`)                                                                     |
| `ssl`           | Enable SSL (from PS 1.7.4)                 | 0                            | 0, 1                                                                                                  |
| `rewrite`       | Enable rewrite engine                      | 1                            | 0, 1                                                                                                  |
| `fixtures`      | Install fixtures                           | 1                            | 0, 1                                                                                                  |

> | `modules`       | Modules to install, separated by comma     | []                           | _array_ of module names (located in `/modules`)                                                       |

Reviewer: What if we leave it blank? It will install all the modules, right?

Author: yes, i added a precision

---
title: Installing PrestaShop from CLI
menuTitle: Installation from CLI
weight: 15
---

# Installing PrestaShop from CLI

> Since version 1.5.4, PrestaShop has had a command-line installer.

Reviewer 2: `PrestaShop has had a command-line installer.` > `PrestaShop has a command-line installer.` ?

Author: Oups, fixed 👍 

```suggestion
Since version 1.5.4, PrestaShop has a command-line installer.
```

This special installer makes it possible to install PrestaShop without the need to use a web browser: simply put the content of the zip archive on your web server or pull code from an official PrestaShop repository, and you can install PrestaShop through your command-line interface (CLI). 

{{% notice info %}}
If deploying from sources (PrestaShop repository), you must first install Composer dependencies.
Install them with `composer install` from project's root directory.
{{% /notice %}}

> The point of having a CLI installer in addition to the regular in-browser installer is to give this option to cater for some advanced users, who often prefer command-line interfaces as they tend to provide a more concise and powerful means to control a program or operating system.

Reviewer: ```suggestion
The point of having a CLI installer in addition to the regular in-browser installer is to provide a solution to some advanced users, who often prefer command-line interfaces as they tend to give a more concise and powerful means to control a program or operating system. We can see the advantage of CLI installer in Continuous Integration processes.
```

## How to use it

> The CLI installer is easy to use: from your terminal, go to the /install (or /install-dev) folder, and start the script with this command:

Reviewer: ```suggestion
To use the CLI installer, use your terminal, go to the `/install` (or `/install-dev`) folder, and start the script with this command:
```

```shell
php index_cli.php
```

> This will display the various available options:

Reviewer: ```suggestion
This command, by default, will display the various available options:
```

```

> Arguments available:

Reviewer: Consider adding a table.

--step	all / database,fixtures,theme,modules,postInstall	(Default: all)

> --language	language iso code	(Default: en)

Reviewer: Worth describing each option. E.g. firstname, name of the administrator, email: used to login to the back office, modules: install all by default or provided as a parameter, etc.

--all_languages	install all available languages	(Default: 0)
--timezone		(Default: Europe/Paris)
--base_uri		(Default: /)
--domain		(Default: localhost)
--db_server		(Default: localhost)
--db_user		(Default: root)
--db_password		(Default: )
--db_name		(Default: prestashop)
--db_clear	Drop existing tables	(Default: 1)
--db_create	Create the database if not exist	(Default: 0)

> --prefix		(Default: ps_)

Reviewer: recommended to change

--engine	InnoDB/MyISAM	(Default: InnoDB)
--name		(Default: PrestaShop)
--activity		(Default: 0)
--country		(Default: fr)
--firstname		(Default: John)
--lastname		(Default: Doe)
--password		(Default: Correct Horse Battery Staple)
--email		(Default: pub@prestashop.com)
--license	show PrestaShop license	(Default: 0)
--theme		(Default: )
--ssl	enable SSL for PrestaShop	(Default: 0)
--rewrite	enable rewrite engine for PrestaShop	(Default: 1)
--fixtures	enable fixtures installation	(Default: 1)
--modules	Modules to install, separated by comma	(Default: [])
```

> - All the options from the regular in-browser installer are available, with their default value listed. 

Reviewer: ```suggestion
- All the options from the regular in-browser installer are available, with their default values listed above.
```

> - Almost all default values value can be left as is, because you can edit them all from the PrestaShop back-office once the installation is done. 

Reviewer: ```suggestion
- Almost all default option values can be left as is because you can edit them all from the PrestaShop back office once the installation is complete. 
```

{{% notice info %}}

> Note that the e-mail and password are the ones used to create the administrator's back-office account.

Reviewer: ```suggestion
Note that the e-mail and password are used to create the administrator's back office account.
```

{{% /notice %}}

> To start the installation, you only need to provide one argument. In reality, you need to provide more:

Reviewer: This does not make sense :D If we need provide more, let's write that we need more 👍

> - `domain`. The location where you want your store to appear.

Reviewer: it's probably not a location as it suggests directory, not a hostname

- `db_server`. The database server address.
- `db_name`. The name of the database you want to use.
- `db_user`. The username for the database you want to use.
- `db_password`. The password for the database username above.

> For instance:

Reviewer: ```suggestion
Example:
```

```shell
php install_cli.php --domain=example.com --db_server=sql.example.com --db_name=prestashop --db_user=root --db_password=123456789
```

> And if your database credentials were good, you should see :

Reviewer: ```suggestion
If the installation completes without any errors, you should see the following confirmation:
```

```shell
-- Installation successful! --
```

{{% notice info %}}

> Please note that your database must be created with a `CREATE DATABASE xxx;` statement before running this command. 

Reviewer: ```suggestion
Before running this command, please note that your database must be created with a `CREATE DATABASE xxx;` statement. 
```

> The Web GUI has an option to **Attempt to create the database automatically** but not the CLI.

Reviewer: ```suggestion
The in-browser installer has an option to attempt to create the database automatically **but not the CLI**.
```
